static/templates/items/melee-details.hbs

```diff
@@ -1,37 +1,46 @@
 <ol class="form-list">
-    {{#if data.damageRolls}}
-        {{#each data.damageRolls as |damageRoll did|}}
-            <li class="damage-part flexrow" data-damage-part="{{did}}">
-                <div class="form-group">
-                    <label>{{localize "PF2E.DamageLabel"}}</label>
-                    <input type="text" name="system.damageRolls.{{did}}.damage" value="{{lookup this "damage"}}" data-dtype="String" style="margin-right:5px"/>
-                </div>
-                <div class="form-group">
-                    <label>{{localize "PF2E.NPCWeaponDamageTypeLabel"}}</label>
-                    <select name="system.damageRolls.{{did}}.damageType" data-dtype="String">
-                        {{#select (lookup this "damageType")}}
-                            {{#each @root.damageTypes as |label type|}}
-                                <option value="{{type}}">{{localize label}}</option>
-                            {{/each}}
-                        {{/select}}
-                    </select>
-                </div>
-                <div style="flex:0; margin-left:5px; display:flex; align-items:center;">
-                    <a class="damage-control delete-damage"><i class="fas fa-minus"></i></a>
-                </div>
-            </li>
-        {{/each}}
-    {{/if}}
-    <div class="form-group" style="display:flex; align-items:center;">
-        <label>
-            <a class="damage-control add-damage">{{localize "PF2E.NPCWeaponAddDamage"}} <i class="fas fa-plus"></i></a>
-        </label>
-    </div>
+    {{#each data.damageRolls as |damageRoll key|}}
+        <li class="damage-component" data-component-key="{{key}}">
+            <div class="form-group">
+                <label>{{localize "PF2E.DamageLabel"}}</label>
+                <input type="text" name="system.damageRolls.{{key}}.damage" value="{{lookup this "damage"}}" />
+            </div>
+            <div class="form-group type">
+                <label>{{localize "PF2E.Roll.Type"}}</label>
+                <select name="system.damageRolls.{{key}}.damageType">
+                    {{#select (lookup damageRoll "damageType")}}
+                        {{#each @root.damageTypes as |label type|}}
+                            <option value="{{type}}">{{localize label}}</option>
+                        {{/each}}
+                    {{/select}}
+                </select>
+            </div>
+            <div class="form-group category">
+                <label>{{localize "PF2E.Category"}}</label>
+                <select name="system.damageRolls.{{key}}.category">
+                    {{#select (lookup damageRoll "category")}}
+                        <option value=""></option>
+                        <option value="persistent">{{localize "PF2E.ConditionTypePersistentShort"}}</option>
+                        <option value="precision">{{localize "PF2E.Damage.Precision"}}</option>
+                        <option value="splash">{{localize "PF2E.TraitSplash"}}</option>
```

Comment:
You should be able to map this from sheet data if you pass in pick(CONFIG.PF2E.damageCategories, DAMAGE_CATEGORIES_UNIQUE). That's what I did for afflictions for example.

Definitely would like a rename for DAMAGE_CATEGORIES_UNIQUE though. That only came about because main was unavailable.